python/hsfs/feature_view.py

```diff
@@ -3486,16 +3513,62 @@ def update_from_response_json(self, json_dict: Dict[str, Any]) -> "FeatureView":
         self._init_feature_monitoring_engine()
         return self
 
-    def transform_batch_data(self, features):
-        return self._feature_view_engine.transform_batch_data(
-            features, self._batch_scoring_server._transformation_functions
+    def compute_on_demand_features(
+        self,
+        feature_vector: Union[List[Any], List[List[Any]], pd.DataFrame, pl.DataFrame],
+        request_parameters: Optional[
+            Union[List[Dict[str, Any]], Dict[str, Any]]
+        ] = None,
+        external: Optional[bool] = None,
+    ):
+        """
+        Function computes on-demand features present in the feature view.
+
+        # Arguments
+            feature_vector: `Union[List[Any], List[List[Any]], pd.DataFrame, pl.DataFrame]`. The feature vector to be transformed.
+            request_parameters: Request parameters required by on-demand transformation functions to compute on-demand features present in the feature view.
+            external: boolean, optional. If set to True, the connection to the
+                online feature store is established using the same host as
+                for the `host` parameter in the [`hsfs.connection()`](connection_api.md#connection) method.
+                If set to False, the online feature store storage connector is used
+                which relies on the private IP. Defaults to True if connection to Hopsworks is established from
+                external environment (e.g AWS Sagemaker or Google Colab), otherwise to False.
+        # Returns
+            `Union[List[Any], List[List[Any]], pd.DataFrame, pl.DataFrame]`: The feature vector that contains all on-demand features in the feature view.
+        """
+        if self._vector_server is None:
+            self.init_serving(external=external)
+
+        return self._batch_scoring_server.compute_on_demand_features(
+            feature_vectors=feature_vector, request_parameters=request_parameters
         )
 
-    def transform_feature_vector(self, features):
-        return self.transform_feature_vectors([features])
+    def transform(
+        self,
+        feature_vector: Union[List[Any], List[List[Any]], pd.DataFrame, pl.DataFrame],
+        external: Optional[bool] = None,
```

Comment:
why is it needed? this function won't fetch features from online store right?

Reply:
Yes you are right it won't fetch features from the online store, but MDT's might require statistics of features. The statistics used while transforming depends on the training_dataset_version set in `init_serving` so if `init_serving` had not been done we don't know the statistics to be used for applying the transformation function. So I basically copied the code done in `get_feature_vector` for initializing vector server. 

I could be wrong here but I think there should be some check to see if `init_serving` or `init_batch_scoring` has been performed.